api/placer.py

```diff
@@ -743,27 +732,45 @@ def process_file_field(self, field, file_attrs):
                     file_attrs.update(file_md)
                     break
 
+        if self.context.get('job_ticket_id'):
+            job_ticket = JobTicket.get(self.context.get('job_ticket_id'))
+
+            if not job_ticket['success']:
+                file_attrs['from_failed_job'] = True
+
         file_attrs['created'] = file_attrs['modified']
         self.save_file(field)
         self.saved.append(file_attrs)
 
     def finalize(self):
-        # Search the sessions table for analysis, replace file field
-        if self.saved:
```

Comment:
Sanity check: you're sure we don't need the `self.saved` conditional? I'm not super familiar here.